### PR TITLE
#3776 Configurable Sync interval

### LINKED ISSUE #3776: Sync time should be configurable
## Is your feature request related to a problem? Please describe.

The sync interval should be configurable

## Describe the solution you'd like

- UI in settings page to set the sync interval
- Not sure where to store this. Maybe a setting

## Implementation

TBD

## Describe alternatives you've considered

N/A

## Additional context

N/A

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #3776 

## Change summary

- Adds configurable sync interval

## Testing

- [ ] Sync interval can be configurable from the settings page

### Related areas to think about

N/A

### CODE REVIEW

File: src/pages/SettingsPage.js [hunk 1]
```diff
@@ -331,6 +169,24 @@ const Settings = ({ toRealmExplorer, currentUserPasswordHash, requestStorageWrit
       <View style={row}>
         <View style={column}>
           <PageInfo columns={pageInfoColumns} />
+          <FlexRow
+            flex={1}
+            justifyContent="center"
+            alignItems="center"
+            style={{ marginTop: 10, maxHeight: 50 }}
```
Comment (mark-prins) [inline src/pages/SettingsPage.js:176]: Couple of style questions.. the font looks different for the new setting, and can we have the slider start where the first two settings start?

settings

**Review by mark-prins — APPROVED** [review]
The default is required when reinstalling - the login screen was locking up for me, eventually crashing with 

```  
ERROR    Warning: Please report: Excessive number of pending callbacks: 501
```

The setting was empty, which would evaluate to an interval of `0` - I think?
I have a recommendation at the bottom for this, which worked for me.

File: src/localization/generalStrings.json [hunk 2]
```diff
@@ -1,5 +1,6 @@
 {
   "gb": {
+    "sync_interval_minutes": "Sync interval (minutes)",
```
Comment (mark-prins) [inline src/localization/generalStrings.json:3]:
```suggestion
    "sync_interval_minutes": "Sync interval (minutes):",
```

**Comment by mark-prins** [comment]
@joshxg I committed the style suggestions in 667cfea4 just in case that is useful. 
No obligation to use! just revert the commit and add in whatever you think is good.

File: src/pages/SettingsPage.js [hunk 3]
```diff
@@ -28,14 +26,16 @@ import { ConfirmIcon } from '../widgets/icons';
...
 import { selectCurrentUserPasswordHash } from '../selectors/user';
 import { PermissionActions } from '../actions/PermissionActions';
-import { createRecord } from '../database/utilities/index';
-import BreachManager from '../bluetooth/BreachManager';
-import { VaccineDataAccess } from '../bluetooth/VaccineDataAccess';
 import { useLoadingIndicator } from '../hooks/useLoadingIndicator';
+import { createVaccineData } from '../database/utilities/dataGenerators';
+import { Slider } from '../widgets/Slider';
+import { FlexRow } from '../widgets/FlexRow';
+import globalStyles, { APP_FONT_FAMILY, DARK_GREY, SUSSOL_ORANGE } from '../globalStyles';
```
Comment (mark-prins) [inline src/pages/SettingsPage.js:36]: Suggestion was to import from `../globalStyles` and not `/colors` as the FONT wasn't getting picked up.

**Comment by joshxg** [comment]
Hmm I feel like I need to test this one more so i'm a little scared to merge ha

**Comment by joshxg** [comment]
Hm lucky- added this one: 42a7cc1 which sets defaults when starting an initialization, which deletes all records. Seems to be working well now

**Pull request merged**